kubernetesserviceapiv1/kubernetes_service_api_v1.go

```diff
@@ -10630,6 +10631,153 @@ func (kubernetesServiceApi *KubernetesServiceApiV1) RemoveSatelliteLocationWithC
+type UpdateSatelliteLocationOptions struct {
+
+	// The ID of the Satellite location.
+	ID *string `json:"id,omitempty"`
+
+	// The name of the Satellite location.
+	Name *string `json:"name,omitempty"`
+
+	// The description of the Satellite location.
+	Description *string `json:"description,omitempty"`
+}
+
+// GraphQLRequest is the JSON request body for a GraphQL HTTP API call
+type GraphQLRequest struct {
+	Query         string         `json:"query"`
+	OperationName string         `json:"operationName,omitempty"`
+	Variables     map[string]any `json:"variables,omitempty"`
+}
+
+// GraphQLResponse is the JSON response body for a GraphQL HTTP API call
+type GraphQLResponse struct {
+	Errors     *GraphQLErrors   `json:"errors,omitempty"`
+	Data       *json.RawMessage `json:"data,omitempty"`
+	Extensions map[string]any   `json:"extensions,omitempty"`
+}
+
+type UpdateSatelliteLocationInput struct {
+	ID          string  `json:"id"`
+	Name        *string `json:"name"`
+	Description *string `json:"description"`
+}
+
+type UpdateSatelliteLocationData struct {
+	UpdateSatelliteLocation UpdateSatelliteLocationPayload `json:"updateSatelliteLocation"`
+}
+
+type UpdateSatelliteLocationPayload struct {
+	SatelliteLocation *SatelliteLocation `json:"satelliteLocation"`
+}
+
+type SatelliteLocation struct {
+	ID          string `json:"id"`
+	Name        string `json:"name"`
+	Description string `json:"description"`
+}
+
+type GraphQLErrors []*GraphQLError
+
+// Error is a GraphQL response error
+type GraphQLError struct {
+	Message    string         `json:"message"`
+	Path       []any          `json:"path,omitempty"`
+	Extensions map[string]any `json:"extensions,omitempty"`
+}
+
+// UpdateSatelliteLocation : Update an IBM Cloud Satellite Location
+// Update an IBM Cloud Satellite location.
+func (kubernetesServiceApi *KubernetesServiceApiV1) UpdateSatelliteLocation(updateSatelliteLocationOptions *UpdateSatelliteLocationOptions) (result UpdateSatelliteLocationPayload, response *core.DetailedResponse, err error) {
+	return kubernetesServiceApi.UpdateSatelliteLocationWithContext(context.Background(), updateSatelliteLocationOptions)
+}
+
+// UpdateSatelliteLocationWithContext is an alternate form of the UpdateSatelliteLocation method which supports a Context parameter
+func (kubernetesServiceApi *KubernetesServiceApiV1) UpdateSatelliteLocationWithContext(ctx context.Context, updateSatelliteLocationOptions *UpdateSatelliteLocationOptions) (result UpdateSatelliteLocationPayload, response *core.DetailedResponse, err error) {
+	err = core.ValidateStruct(updateSatelliteLocationOptions, "updateSatelliteLocationOptions")
+	if err != nil {
+		return
+	}
+
+	builder := core.NewRequestBuilder(core.POST)
+	builder = builder.WithContext(ctx)
+	builder.EnableGzipCompression = kubernetesServiceApi.GetEnableGzipCompression()
+	_, err = builder.ResolveRequestURL(kubernetesServiceApi.Service.Options.URL, `/unsafe/graphql`, nil)
```

Comment:
This is an unsafe route. We can't ship this to terraform as we'd have to guarantee that we do not change this route ever.